### PR TITLE
Remove scheduled save when during asset deletion, closes #90

### LINKED ISSUE #90: Deleting an asset with a scheduled save doesn't cancel scheduled save
It looks like deleting an asset with a scheduled save doesn't cancel the scheduled save, leading to ENOENT error messages in the console.

To reproduce:

 1. Create a sprite
 2. Upload a texture
 3. Within a minute, trash it
 4. Wait 60 seconds

### PR DESCRIPTION
**Pull request by chrisbubernak**
Added code inside the asset deletion method to remove the pending callback from the scheduled save.

### CODE REVIEW

**Comment by chrisbubernak** [comment]
So my fix stops the callback from happening but I noticed that the object containing the scheduled saves never actually has entries removed from it until the server goes offline, all that happens is the values in it get null-ed out. Is this intentional? Seem's kind of like a memory leak in a sense because in the pathological case (adding tons of sprites /w images to your project) this is just going to grow and hang around for ever. Is there some purpose for that? If not, is it something work fixing?

File: server/RemoteProjectClient.ts [hunk 1]
```diff
@@ -215,6 +215,14 @@ export default class RemoteProjectClient extends BaseRemoteClient {
                 }
               }
 
+              // Cancel any scheduled save for this asset
+              let scheduledSaveCallback = this.server.scheduledSaveCallbacks[`assets:${entry.id}`];
+              if (scheduledSaveCallback) {
```
Comment (elisee) [inline server/RemoteProjectClient.ts]: Prefer `(scheduledSaveCallback != null)`

File: server/RemoteProjectClient.ts [hunk 2]
```diff
@@ -215,6 +215,14 @@ export default class RemoteProjectClient extends BaseRemoteClient {
                 }
               }
 
+              // Cancel any scheduled save for this asset
+              let scheduledSaveCallback = this.server.scheduledSaveCallbacks[`assets:${entry.id}`];
```
Comment (elisee) [inline server/RemoteProjectClient.ts]: Prefer `const` over `let` if variable won't be reassigned

**Comment by elisee** [comment]
Gave this some thought. Since assets on disk will be preserved in the `trashedAssets` folder, I feel like we should not cancel the scheduled save, but rather force it to happen right away (before or after moving the folder elsewhere)? What does everyone think?

> So my fix stops the callback from happening but I noticed that the object containing the scheduled saves never actually has entries removed from it until the server goes offline, all that happens is the values in it get null-ed out. Is this intentional?

Yeah that looks wrong indeed. I can't think of a reason why we would rather null them out than deleting them.

**Comment by chrisbubernak** [comment]
> Gave this some thought. Since assets on disk will be preserved in the trashedAssets folder, I feel like we should not cancel the scheduled save, but rather force it to happen right away (before or after moving the folder elsewhere)?

@elisee just  to clarify we still want to remove the save from the scheduledsaves object but we just want to actually do the save then instead of just forgetting about it, correct?

**Comment by elisee** [comment]
@chrisbubernak Yes that sounds like the correct approach. But since saves are async operations, we probably need to move the asset folder first and ensure that the save happens in the trashedAssets path.

(This probably all ties in with https://github.com/superpowers/superpowers-core/issues/67 but we should not worry too much about it for now)

**Comment by bilou84** [comment]
So we are not deleting the scheduledCallback entry because it also store the last time it was saved.
But in the case of trashing the entry, I think it's fine to deleted it.

**Pull request closed without merging**